From 16b26154142b10e1f9ced9e0de62956585eb83bd Mon Sep 17 00:00:00 2001
From: Bharat Bhushan <Bharat.Bhushan@freescale.com>
Date: Fri, 4 Mar 2016 19:56:00 +0530
Subject: [PATCH 08/15] KVM arm: Increase the Memslots and expose to usespace

We are direct assigning the the fsl-mc container which have
many devices and these devices have memory regions.
We are also working on direct assigning these fsl-mc containers
to userspace in VM, example enabling ODP/DPDK in guest VM.

So overall the number of device memory regions are more and
32 memslots are not sufficient. So this patch increases the
number of memslots from 32 to 54.

Also this patch add support of exposing these memslots count
to user-sapce.

Signed-off-by: Bharat Bhushan <Bharat.Bhushan@freescale.com>
---
 arch/arm/kvm/arm.c                | 3 +++
 arch/arm64/include/asm/kvm_host.h | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm/kvm/arm.c b/arch/arm/kvm/arm.c
index adb1988..64b6202 100644
--- a/arch/arm/kvm/arm.c
+++ b/arch/arm/kvm/arm.c
@@ -196,6 +196,9 @@ int kvm_vm_ioctl_check_extension(struct kvm *kvm, long ext)
 	case KVM_CAP_MAX_VCPUS:
 		r = KVM_MAX_VCPUS;
 		break;
+	case KVM_CAP_NR_MEMSLOTS:
+		r =  KVM_USER_MEM_SLOTS;
+		break;
 	default:
 		r = kvm_arch_dev_ioctl_check_extension(ext);
 		break;
diff --git a/arch/arm64/include/asm/kvm_host.h b/arch/arm64/include/asm/kvm_host.h
index f0f58c9..de5dfb1 100644
--- a/arch/arm64/include/asm/kvm_host.h
+++ b/arch/arm64/include/asm/kvm_host.h
@@ -36,7 +36,7 @@
 #define KVM_MAX_VCPUS 0
 #endif
 
-#define KVM_USER_MEM_SLOTS 32
+#define KVM_USER_MEM_SLOTS 64
 #define KVM_PRIVATE_MEM_SLOTS 4
 #define KVM_COALESCED_MMIO_PAGE_OFFSET 1
 
-- 
1.8.3.2

